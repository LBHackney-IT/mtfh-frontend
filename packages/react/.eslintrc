{
  "extends": "@lbh-mtfh/eslint-config",
  "parserOptions": {
    "project": "tsconfig.json"
  },
  "rules": {
    "prefer-arrow-callback": "off",
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["**/*/rollup.config.js"]
      }
    ],
    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": ["Link"],
        "aspects": ["invalidHref"]
      }
    ]
  }
}
