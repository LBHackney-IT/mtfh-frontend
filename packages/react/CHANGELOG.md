# @lbh-mtfh/react

## 1.1.0

### Minor Changes

- 37b7b3e: Initial release

### Patch Changes

- Updated dependencies [37b7b3e]
  - @lbh-mtfh/system@1.1.0
  - @lbh-mtfh/test-utils@1.1.0
