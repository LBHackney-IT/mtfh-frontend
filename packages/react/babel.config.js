const babel = require("../../babel.config")

module.exports = babel
