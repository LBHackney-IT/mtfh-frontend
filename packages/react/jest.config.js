const baseConfig = require("../../jest.config")

baseConfig.setupFilesAfterEnv.push("<rootDir>/jest.setup.ts")

module.exports = {
  ...baseConfig,
}
