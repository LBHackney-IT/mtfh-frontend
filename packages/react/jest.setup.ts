import * as LBHFrontend from "lbh-frontend"

window.LBHFrontend = LBHFrontend
