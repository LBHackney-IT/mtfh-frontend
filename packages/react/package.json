{
  "name": "@lbh-mtfh/react",
  "version": "1.0.0",
  "description": "React component library of the LBH Design System",
  "author": "https://hackney.gov.uk/",
  "homepage": "https://github.com/LBHackney-IT/mtfh-frontend",
  "license": "MIT",
  "sideEffects": false,
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/esm/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/cjs/index.js",
      "default": "./dist/esm/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/LBHackney-IT/mtfh-frontend.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "rollup -c"
  },
  "bugs": {
    "url": "https://github.com/LBHackney-IT/mtfh-frontend/issues"
  },
  "devDependencies": {
    "@lbh-mtfh/eslint-config": "^1.0.0",
    "@rollup/plugin-commonjs": "^19.0.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "@rollup/plugin-typescript": "^8.2.1",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@testing-library/user-event": "^13.1.9",
    "axe-core": "^4.2.3",
    "axios": "^0.21.1",
    "jest": "^27.0.6",
    "jest-axe": "^5.0.1",
    "node-sass": "^6.0.1",
    "postcss": "^8.3.5",
    "rollup": "^2.52.8",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.0"
  },
  "dependencies": {
    "@lbh-mtfh/system": "^1.0.0",
    "@lbh-mtfh/test-utils": "^1.0.0",
    "@mtfh/common": "https://github.com/LBHackney-IT/mtfh-frontend-common.git",
    "@radix-ui/react-polymorphic": "^0.0.12",
    "@reach/accordion": "^0.15.3",
    "@reach/dialog": "^0.15.3",
    "@types/lbh-frontend": "^1.0.0",
    "classnames": "^2.3.1",
    "generic-names": "^3.0.0",
    "govuk-frontend": "^3.13.0",
    "lbh-frontend": "^3.5.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-merge-refs": "^1.1.0",
    "swr": "^0.5.6",
    "use-breakpoint": "^2.0.1"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
