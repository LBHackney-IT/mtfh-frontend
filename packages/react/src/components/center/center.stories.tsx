import React from "react";

import { Center } from "./center";

export default {
  title: "Center",
  component: Center,
};

const Dot = () => (
  <div
    style={{
      width: "50px",
      height: "50px",
      background: "green",
      borderRadius: "50%",
    }}
  />
);

export const Base = () => {
  return (
    <Center>
      <Dot />
    </Center>
  );
};

export const Vertical = () => {
  return (
    <Center horizontally={false} style={{ height: "200px" }}>
      <Dot />
    </Center>
  );
};

export const Horizontal = () => {
  return (
    <Center vertically={false} style={{ height: "200px" }}>
      <Dot />
    </Center>
  );
};
