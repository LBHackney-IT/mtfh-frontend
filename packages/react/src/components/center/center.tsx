import React, { forwardRef } from "react";
import type * as Polymorphic from "@radix-ui/react-polymorphic";
import cn from "classnames";

import { widthOverrides } from "../../utils";
import styles from "./styles.module.scss";

export interface CenterProps {
  horizontally?: boolean;
  vertically?: boolean;
  override?: number;
}

export type CenterComponent = Polymorphic.ForwardRefComponent<
  "div",
  CenterProps
>;

export const Center: CenterComponent = forwardRef(function Center(
  {
    as: CenterComp = "div",
    horizontally = true,
    vertically = true,
    className,
    override,
    ...props
  },
  ref
) {
  return (
    <CenterComp
      ref={ref}
      className={cn(
        styles.center,
        {
          [styles.centerHorizontal]: horizontally,
          [styles.centerVertical]: vertically,
        },
        widthOverrides(override),
        className
      )}
      {...props}
    />
  );
});
