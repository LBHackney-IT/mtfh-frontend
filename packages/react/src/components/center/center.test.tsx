import React from "react";
import { render } from "@testing-library/react";

import { testA11y } from "@lbh-mtfh/test-utils";
import { Center } from "./center";

test("it renders with both vertical and horizontal by default", async () => {
  const { container } = render(
    <Center>
      <div />
    </Center>
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders with vertical only", () => {
  const { container } = render(
    <Center horizontally={false}>
      <div />
    </Center>
  );
  expect(container).toMatchSnapshot();
});

test("it renders with horizontal only", () => {
  const { container } = render(
    <Center vertically={false}>
      <div />
    </Center>
  );
  expect(container).toMatchSnapshot();
});
