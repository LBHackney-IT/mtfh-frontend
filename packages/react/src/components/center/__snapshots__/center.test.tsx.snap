// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders with both vertical and horizontal by default 1`] = `
<div>
  <div
    class="center centerHorizontal centerVertical"
  >
    <div />
  </div>
</div>
`;

exports[`it renders with horizontal only 1`] = `
<div>
  <div
    class="center centerHorizontal"
  >
    <div />
  </div>
</div>
`;

exports[`it renders with vertical only 1`] = `
<div>
  <div
    class="center centerVertical"
  >
    <div />
  </div>
</div>
`;
